test/middleware_test.coffee

```diff
@@ -337,7 +349,29 @@ describe 'Middleware', ->
             )
             testDone()
 
+    describe 'receive middleware context', ->
+      beforeEach ->
+        @robot.receiveMiddleware (context, next, done) =>
+          @middleware.call @, context, next, done
+
+      describe 'response', ->
+        it 'is a match-less Response object', (testDone) ->
+          @robot.receive @testMessage, () =>
+            expect(@middleware).to.have.been.calledWithMatch(
+              sinon.match.has('response',
+                sinon.match.instanceOf(Response).and(
+                  sinon.match.has('message',
+                    sinon.match.same(@testMessage)))) # context
+              sinon.match.any                         # next
+              sinon.match.any                         # done
+            )
+            testDone()
+
     describe 'next', ->
+      beforeEach ->
+        @robot.listenerMiddleware (context, next, done) =>
```

Comment:
I saw this as an integration test, and worth having, but I think the robot tests are the right place to have full coverage for the different kinds of middleware (and I added a bunch of tests there).